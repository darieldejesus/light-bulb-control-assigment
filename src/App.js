import React from 'react';
import {
  Container,
  Flex,
  Box,
} from 'rendition';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import styled from 'styled-components';


import Navbar from './components/Navbar';
import DevicesTable from './components/DevicesTable';
import reducers from './stateManager/reducers';
import clientMiddleware from './stateManager/middleware/client';
import api from './helpers/ApiClient';
import BrightnessControl from './components/BrightnessControl';
import NotificationBar from './components/NotificationBar';
import ActionBar from './components/ActionBar';

/*
 * Setup Redux Store with the Middleware.
 * */
const reduxStore = createStore(
  reducers,
  applyMiddleware(
    clientMiddleware(api),
    thunk,
  ),
);

const BrightnessControlWrapper = styled(Box)`
  border-radius: 10px;
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
`;

const AppWrapper = styled.div`
  height: 100%;
  min-height: 100%;
`;

export default () => (
  <Provider store={reduxStore}>
    <AppWrapper>
      <Navbar />
      <ActionBar />
      <Container>
        <NotificationBar />
        <Flex flexWrap="wrap-reverse">
          <Box width={[1, 1, 1 / 2, 2 / 3]} pr={[0, 0, 3]}>
            <DevicesTable />
          </Box>
          <BrightnessControlWrapper mb={[3, 0]} width={[1, 1, 1 / 2, 1 / 3]} bg="tertiary.dark">
            <BrightnessControl />
          </BrightnessControlWrapper>
        </Flex>
      </Container>
    </AppWrapper>
  </Provider>
);
