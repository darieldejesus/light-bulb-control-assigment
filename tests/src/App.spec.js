import React from 'react';

import App from '../../src/App';

describe('App', () => {
  test('Should render the main page', () => {
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
