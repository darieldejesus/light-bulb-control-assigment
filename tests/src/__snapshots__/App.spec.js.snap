// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render the main page 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <styled.div>
    <Navbar />
    <Component />
    <WithTheme(Component)>
      <Connect(NotificationBar) />
      <Component
        flexWrap="wrap-reverse"
      >
        <Component
          pr={
            Array [
              0,
              0,
              3,
            ]
          }
          width={
            Array [
              1,
              1,
              0.5,
              0.6666666666666666,
            ]
          }
        >
          <Connect(DevicesTable) />
        </Component>
        <Styled(Component)
          bg="tertiary.dark"
          mb={
            Array [
              3,
              0,
            ]
          }
          width={
            Array [
              1,
              1,
              0.5,
              0.3333333333333333,
            ]
          }
        >
          <Connect(BrightnessControl) />
        </Styled(Component)>
      </Component>
    </WithTheme(Component)>
  </styled.div>
</Provider>
`;
